File: client-credentials-flow-aws-cdk/lib/auth_code_flow_stack.py
from aws_cdk import (
    core
)

from lib.api_gateway_stack import ApiGatewayStack
from lib.cognito_stack import CognitoStack


class ClientCredsFlowStack(core.Stack):
    def __init__(self,
                 scope: core.Construct,
                 id: str,
                 **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        cognito_stack = CognitoStack(self, "CognitoStack")

        api_gateway_stack = ApiGatewayStack(self, "ApiGatewayStack", cognito_stack.cognito_user_pool)
        api_gateway_stack.add_dependency(cognito_stack)

File: authorization-code-flow/lib/auth-code-flow-stack.ts
import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { CognitoStack } from './cognito-stack';
import { ApiGatewayStack } from './api-gateway-stack';

export class AuthCodeFlowStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const cognitoStack = new CognitoStack(this, 'CognitoStack');
    const apiGatewayStack = new ApiGatewayStack(this, 'ApiGatewayStack', cognitoStack.cognitoUserPool);
    apiGatewayStack.addDependency(cognitoStack);
  }
}
